src/modules/PublicLab.RichTextModule.js

```diff
@@ -310,15 +310,15 @@ module.exports = PublicLab.RichTextModule = PublicLab.Module.extend({
     var wkC = document.getElementsByClassName("wk-commands")[0];
 
     $(window).scroll(function() {
-      var textAreaRect = document
-          .getElementsByClassName("wk-container")[0]
+      var textAreaRect = $(".ple-textarea")[0]
           .getBoundingClientRect();
-      var footerRect = document
-          .getElementsByClassName("ple-footer")[0]
+      var richAreaRect = $(".wk-wysiwyg")[0]
+          .getBoundingClientRect();
+      var footerRect = $("ple-footer")[0]
           .getBoundingClientRect().height;
 
       if (
-        textAreaRect.bottom >= ((window.innerHeight || document.documentElement.clientHeight) - footerRect) && textAreaRect.top <= ((window.innerHeight || document.documentElement.clientHeight) - footerRect)
+        (textAreaRect.bottom || richAreaRect.bottom) >= ((window.innerHeight || document.documentElement.clientHeight) - footerRect) && (textAreaRect.top || richAreaRect.top) <= ((window.innerHeight || document.documentElement.clientHeight) - footerRect)
```

Comment:
Awesome. OK, now let's see why it's not passing tests. Maybe it was just a timeout or something.

Reply:
Aha, can we actually do a comparison nested like this? `(textAreaRect.top || richAreaRect.top) <= (` ? I think we need to split the comparison in 2... i'm just not familiar with this kind of nesting, you may be way ahead of me here... or, possibly, could it be that this is not working in the version of JS/ES or Node that we are using?

Reply:
Hi @Shulammite-Aso could you clarify this just so we can follow the code? It's still a bit confusing to me. Thank you!!!

Reply:
Hi @jywarren , So this piece `(textAreaRect.top || richAreaRect.top) <= (` is making the toolbar float for both rich and markdown editing, and it's actually working fine.  I think splitting the comparison in 2 will cause a lot of redundancy, something like this?
`if (
        (textAreaRect.bottom >= ((window.innerHeight || document.documentElement.clientHeight) - footerRect) && textAreaRect.top <= ((window.innerHeight || document.documentElement.clientHeight) - footerRect)) || (richAreaRect.bottom >= ((window.innerHeight || document.documentElement.clientHeight) - footerRect) && richAreaRect.top <= ((window.innerHeight || document.documentElement.clientHeight) - footerRect))
      ) {`
plus it will still actually be nested.  
And not nesting the comparison at all like this `textAreaRect.bottom || richAreaRect.bottom >=` is not producing the wanted result.

I was using `wk-container` , the div that was holding both the `textarea` for markdown and `contenteditable` div for wysiwyg to acheive this result before, but this didn't look too good from plots2, as the  `contenteditable` div is way shorter than `wk-container` div, making the toolbar still begin to float a little before the  `contenteditable` div